gatsby/pages.rst

```diff
@@ -16,64 +24,62 @@ If we see the file structure of our `hello-world` project, we can see that there
     │       └── page-2.js
     ...
 
-Let's see how is made `index.js` file. This page is the homepage of our example site.
+Let's see how is made ``index.js`` file. This page is the homepage of our example site.
 
-.. code-block:: none
+.. literalinclude:: _snippets/index_orig.js
+  :language: jsx
 
-  import React from 'react'
-  import { Link } from 'gatsby'  
+As we said previously, pages are not basic HTML documents, but they are React components.
 
-  const IndexPage = () => (
-    <div>
-      <h1>Gatsby Site</h1>
-      <p>Welcome to your new Gatsby site.</p>
-      <p>Now go build something great.</p>
-      <Link to="/page-2/">Go to page 2</Link>
-    </div>
-  )  
-  export default IndexPage
+You can see that React components are written in a particular syntax called ``JSX``.
 
-As we said previously, pages are not basic html documents, but they are React components.
-You can see that React components are written in a particular syntax called `JSX` that allows to mix pure javascript with
-some html tags.
-Components are functions (or es6 classes) that accept some data and renders some html.
+``JSX`` allows us to mix pure JavaScript with some HTML tags.
 
-.. note:: You can see more infos in the official `ReactJS documentation <https://reactjs.org/docs/components-and-props.html>`_
+Components are functions (or ES6 classes) that accept some data and renders some HTML.
```

Comment:
Above 4 sentences can be 1 paragraph by removing blank lines that separate them.